internal/server/manager.go

```diff
@@ -152,6 +190,42 @@ func (m *instance) RemoveCAPICluster(clusterNamespace, clusterName string) {
 	delete(m.capiClusters, *clusterInfo)
 }
 
+func (m *instance) AddClusterProfileStatus(summary *configv1alpha1.ClusterSummary) {
+	if !isClusterSummary(summary) {
+		return
+	}
+
+	// we're sure we're adding a proper cluster summary
+	// get the cluster profile name by using labels
+	// TODO: I didn't get where to get this cluster-profile label name (same as in isClusterSummary)...
```

Comment:
to get (Cluster)Profile name (and verify it is properly set) just call

profileOwnerRef, err := configv1alpha1.GetProfileOwnerReference(summary)